test/steps/kogito_service_table_parser.go

```diff
@@ -0,0 +1,134 @@
+)
+
+// *** Whenever you add new parsing functionality here please add corresponding DataTable example to every file in steps which can use the functionality ***
+
+const (
+	// DataTable first column
+	kogitoServiceInfinispanKey      = "infinispan"
+	kogitoServiceKafkaKey           = "kafka"
+	kogitoServiceRuntimeRequestKey  = "runtime-request"
+	kogitoServiceRuntimeLimitKey    = "runtime-limit"
+	kogitoServiceRuntimeEnvKey      = "runtime-env"
+	kogitoServiceServiceLabelKey    = "service-label"
+	kogitoServiceDeploymentLabelKey = "deployment-label"
+
+	// DataTable second column
+	kogitoServiceInfinispanUseKogitoInfraKey = "useKogitoInfra"
+	kogitoServiceInfinispanUsernameKey       = "username"
+	kogitoServiceInfinispanPasswordKey       = "password"
+	kogitoServiceInfinispanURIKey            = "uri"
+	kogitoServiceKafkaUseKogitoInfraKey      = "useKogitoInfra"
+	kogitoServiceKafkaExternalURIKey         = "externalURI"
+	kogitoServiceKafkaInstanceKey            = "instance"
+)
+
+func configureKogitoServiceFromTable(table *messages.PickleStepArgument_PickleTable, kogitoService *framework.KogitoServiceHolder) (err error) {
+	if len(table.Rows) == 0 { // Using default configuration
+		return nil
+	}
+
+	if len(table.Rows[0].Cells) != 3 {
+		return fmt.Errorf("expected table to have exactly three columns")
+	}
+
+	for _, row := range table.Rows {
+		firstColumn := getFirstColumn(row)
+
+		switch firstColumn {
+		case kogitoServiceInfinispanKey:
+			err = parseKogitoServiceInfinispanRow(row, kogitoService)
+
+		case kogitoServiceKafkaKey:
+			err = parseKogitoServiceKafkaRow(row, kogitoService)
+
+		case kogitoServiceServiceLabelKey:
+			kogitoService.KogitoService.GetSpec().GetServiceLabels()[getSecondColumn(row)] = getThirdColumn(row)
+
+		case kogitoServiceDeploymentLabelKey:
+			kogitoService.KogitoService.GetSpec().GetDeploymentLabels()[getSecondColumn(row)] = getThirdColumn(row)
+
+		case kogitoServiceRuntimeEnvKey:
+			kogitoService.KogitoService.GetSpec().AddEnvironmentVariable(getSecondColumn(row), getThirdColumn(row))
+
+		case kogitoServiceRuntimeRequestKey:
+			kogitoService.KogitoService.GetSpec().AddResourceRequest(getSecondColumn(row), getThirdColumn(row))
+
+		case kogitoServiceRuntimeLimitKey:
+			kogitoService.KogitoService.GetSpec().AddResourceLimit(getSecondColumn(row), getThirdColumn(row))
+
+		default:
+			err = fmt.Errorf("Unrecognized configuration option: %s", firstColumn)
+		}
+
+		if err != nil {
+			return
+		}
+	}
+
+	return
+}
+
+func parseKogitoServiceInfinispanRow(row *messages.PickleStepArgument_PickleTable_PickleTableRow, kogitoService *framework.KogitoServiceHolder) error {
+	secondColumn := getSecondColumn(row)
+
+	if infinispanAware, ok := kogitoService.KogitoService.GetSpec().(v1alpha1.InfinispanAware); ok {
+		switch secondColumn {
+		case kogitoServiceInfinispanUseKogitoInfraKey:
+			infinispanAware.GetInfinispanProperties().UseKogitoInfra = framework.MustParseEnabledDisabled(getThirdColumn(row))
```

Comment:
Just a note that this option cannot be combined with other options when Infinispan is set manually, right?

Reply:
Well, IMHO it shouldn't be combined.
Though it is possible to define both in CR, in that case I am not sure what will happen (may be worth trying).

Reply:
So, it seems that if CLI is used with both Infinispan options then external Infinispan is used, KogitoInfra one is ignored. In case CR is used then it is vice versa, KogitoInfra is used and external config is ignored.

Reply:
Interesting :)
Anyway, it is a misconfiguration so we probably don't need to address it. Although maybe a warning would be great from operator side that user/password is set even though KogitoInfra is specified.